pkg/reconciler/openshift/tektonaddon/tektonaddon.go

```diff
@@ -369,26 +354,34 @@ func (r *Reconciler) ensureClusterTasks(ctx context.Context, ta *v1alpha1.Tekton
 // and if installer set which already exist is of older version then it deletes and return false to create a new
 // installer set
 func checkIfInstallerSetExist(ctx context.Context, oc clientset.Interface, relVersion string,
-	ta *v1alpha1.TektonAddon, component string) (bool, error) {
+	component string) (bool, error) {
 
-	// Check if installer set is already created
-	compInstallerSet, ok := ta.Status.AddonsInstallerSet[component]
-	if !ok {
-		return false, nil
+	installerSets, err := oc.OperatorV1alpha1().TektonInstallerSets().
+		List(ctx, metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("%s=%s", tektoninstallerset.InstallerSetType, component),
+		})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return false, nil
+		}
+		return false, err
 	}
 
-	if compInstallerSet != "" {
-		// if already created then check which version it is
-		ctIs, err := oc.OperatorV1alpha1().TektonInstallerSets().
-			Get(ctx, compInstallerSet, metav1.GetOptions{})
+	// Delete installer sets if exist more than one
+	if len(installerSets.Items) > 1 {
+		err = oc.OperatorV1alpha1().TektonInstallerSets().
+			DeleteCollection(ctx, metav1.DeleteOptions{}, metav1.ListOptions{
+				LabelSelector: fmt.Sprintf("%s=%s", tektoninstallerset.InstallerSetType, component),
+			})
 		if err != nil {
-			if errors.IsNotFound(err) {
-				return false, nil
-			}
 			return false, err
 		}
+	}
 
-		version, ok := ctIs.Annotations[tektoninstallerset.ReleaseVersionKey]
+	// Check if installer set is already created
+	for i := range installerSets.Items {
```

Comment:
what is the purpose of this `for loop`. 
the above `DeleteCollection()` call will send a `delete `  api call so why do we try to process the same set with this `for loop`.

All the items in this  instalerset.Items is being deleted because of the above call, isn't it.

Reply:
the DeleteCollection call only makes an Api call, it won't make `installerSets.Items` empty.